### PR TITLE
SW-1537 Calculate observed quantity in v1->v2 conversion

### PR DESCRIPTION
**Pull request by sgrimm**
If you read an accession with the v2 API and it was written to the database with
the v1 API, or vice versa, we do an on-the-fly conversion to the appropriate
style.

Previously, this was done in the controller and payload classes, but as the new
quantity tracking logic rolls out, the conversion is going to get complex enough
and involve enough business logic that it makes more sense to do it closer to the
rest of the business logic. Move the calculation to `AccessionModel` and add
fields to hold the observed quantity and time.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #586** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/586" target="_blank">Graphite</a> 
    * **PR #587** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/587" target="_blank">Graphite</a>  👈
      * **PR #588** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/588" target="_blank">Graphite</a> 
        * **PR #589** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/589" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/587?utm_source=stack-comment).

File: src/main/kotlin/com/terraformation/backend/seedbank/model/Accession.kt [hunk 1]
```diff
@@ -161,6 +145,24 @@ data class AccessionModel(
...
+    } else {
+      copy(isManualState = false).withCalculatedValues(clock)
+    }
+  }
+
+  fun toV2Compatible(clock: Clock): AccessionModel {
+    return if (isManualState) {
+      this
+    } else {
+      withCalculatedValues(clock)
```
Comment (karthikbtf) [inline src/main/kotlin/com/terraformation/backend/seedbank/model/Accession.kt:164]: is this first withCalculatedValues intentional?

Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/seedbank/model/Accession.kt:164]: Yeah, but good point that it's not obvious why. Added a comment explaining what it's for.

Comment (karthikbtf) [reply src/main/kotlin/com/terraformation/backend/seedbank/model/Accession.kt:164]: Thanks! Makes sense, had a rough idea.

**Review by karthikbtf — APPROVED** [review]

**Comment by sgrimm** [comment]
**Graphite Merge Job**

Current status: ✅ Merged

This pull request was successfully merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/587?utm_source=stack-comment).


Job Reference: QpIP1QCR1Nvluk7Y6jyS

**Pull request merged**